poetry_dynamic_versioning/__init__.py

```diff
@@ -173,9 +173,13 @@ def _get_version(config: Mapping) -> Tuple[_DUNAMAI_VERSION_ANY, str]:
     if style is not None:
         style = Style(style)
 
-    version = Version.from_vcs(
-        vcs, config["pattern"], config["latest-tag"], config["subversion"]["tag-dir"]
-    )
+    # allow for version to be overwritten, or fetch from vcs
+    if "POETRY_DYNAMIC_VERSIONING_PRETEND_VERSION" in os.environ.keys:
+        version = Version(os.environ.get("POETRY_DYNAMIC_VERSIONING_PRETEND_VERSION"))
```

Comment:
To make this work, we should do a little bit of refactoring:

`_get_version()` currently returns both a Dunamai  `Version` and an opaque string, but the `Version` isn't really needed (it's stored, but not being used currently) and is causing a complication for this new option. I think `_get_version()` should just immediately return the opaque string from the environment variable if it's set and ditch the `Version` from the return.

Since the environment variable is a backdoor, we should make sure we skip the formatting/bumping logic and use it as-is. In other words, rather than setting `1.2.3` in the env var and letting the plugin bump it to `1.2.4`, I would expect to set the env var to `1.2.4` directly.